autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -44,42 +62,86 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
         }
         LoggingUtils.buildMdcContext(cluster);
 
-        StackStatusV4Response statusResponse = cloudbreakCommunicator.getStackStatusByCrn(cluster.getStackCrn());
+        StackV4Response stackResponse = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
+        Set<String> policyHostGroups = extractPolicyHostGroups(cluster);
+        DependentHostGroupsV4Response dependentHostGroupsResponse = dependentHostGroupsService.getDependentHostGroupsForPolicyHostGroup(cluster.getStackCrn(),
+                policyHostGroups);
 
-        boolean clusterAvailable;
-        if (Boolean.TRUE.equals(cluster.isStopStartScalingEnabled())) {
-            clusterAvailable = Optional.ofNullable(statusResponse.getStatus()).map(Status::isAvailable).orElse(false);
-            // TODO CB-15146: This may need to change depending on the final form of how we check which operations are to be allowed
-            //  when there are some STOPPED instances
-        } else {
-            clusterAvailable = Optional.ofNullable(statusResponse.getStatus()).map(Status::isAvailable).orElse(false)
-                    && Optional.ofNullable(statusResponse.getClusterStatus()).map(Status::isAvailable).orElse(false);
-        }
+        boolean cmAvailable = cmCommunicator.isClusterManagerRunning(cluster);
```

Comment:
Thought we were not going to reach out to CM, and instead rely on the status returned by Cloudbreak for the CM nodes. This should be OK though, since CM has to be available, and we must be able to reach it.

What is the retry and timeout policy on this particular invocation ? - How long can it cause a cluster to get stuck.

Also, does the CB response provide information on the 'type' - core vs gateway. 'gateway' is where CM ends up running, and is limited to a single host group.

Reply:
The retry and timeout durations are set to 125 seconds defined in periscope configs - `application.yml`, which can be shortened if required. CB does provide this info but I thought this approach is more accurate, since we don't have to parse through the CB stackResponse to identify the CM node and infer its health